controllers/topology/cluster_controller_test.go

```diff
@@ -283,51 +286,143 @@ func TestClusterReconciler_reconcileUpdatesOnClusterClass(t *testing.T) {
 	}, timeout).Should(Succeed())
 }
 
+func TestClusterReconciler_reconcileClusterClassRebase(t *testing.T) {
+	defer utilfeature.SetFeatureGateDuringTest(t, feature.Gates, feature.ClusterTopology, true)()
+	g := NewWithT(t)
+	timeout := 300 * time.Second
+
+	ns, err := env.CreateNamespace(ctx, "test-topology-cluster-reconcile")
+	g.Expect(err).ToNot(HaveOccurred())
+
+	// Create the objects needed for the integration test:
+	// - a ClusterClass with all the related templates
+	// - a Cluster using the first ClusterClass
+	// - a compatible ClusterClass to rebase the Cluster to
+	cleanup, err := setupTestEnvForIntegrationTests(ns)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	// Defer a cleanup function that deletes each of the objects created during setupTestEnvForIntegrationTests.
+	defer func() {
+		g.Expect(cleanup()).To(Succeed())
+	}()
+
+	actualCluster := &clusterv1.Cluster{}
+	// First ensure that the initial cluster and other objects are created and populated as expected.
+	g.Eventually(func(g Gomega) error {
+		// Get the cluster object now including the updated replica number for the Machine deployment.
```

Comment:
```suggestion
		// Get the cluster object.
```
nit: (not sure if my comment is 100% correct copy&pasted it from another test)